src/applications/appeals/shared/validations/contactInfo.js

```diff
@@ -1,14 +1,50 @@
 import { errorMessages } from '../content/contactInfo';
 
+// HLR & NOD validation; has homeless question & only mobile phone
 export const contactInfoValidation = (errors = {}, _fieldData, formData) => {
   const { veteran = {}, homeless } = formData || {};
+  if (!veteran.phone?.phoneNumber) {
+    errors.addError?.(errorMessages.missingMobilePhone);
+  }
   if (!veteran.email) {
     errors.addError?.(errorMessages.missingEmail);
   }
-  if (!veteran.phone?.phoneNumber) {
+  if (!homeless) {
+    const address = veteran.address || {};
+    if (!address.addressLine1) {
+      errors.addError?.(errorMessages.missingMailingAddress);
+    }
+    if (
+      !address.city ||
+      (address.countryCodeIso2 === 'US' &&
+        (!address.stateCode || address.zipCode?.length !== 5))
+    ) {
+      errors.addError?.(errorMessages.invalidMailingAddress);
+    }
+  }
+};
+
+// 995 validation; no homeless question & has both home & mobile phones
+export const contactInfo995Validation = (errors = {}, _fieldData, formData) => {
```

Comment:
Is there any way to refactor this into the method above?

Reply:
Hmmm, I could write this, but I feel like it's harder to read. WDYT?

```js
export const contactInfo995Validation = (errors = {}, _fieldData, formData) => {
  const { veteran = {} } = formData || {};
  if (!veteran?.homePhone?.phoneNumber) {
    errors.addError?.(errorMessages.missingHomePhone);
  }
  // Set homeless to false so address check is performed
  const newFormData = {
    ...formData,
    homeless: false,
    veteran: {
      ...veteran,
      // mobilePhone becomes phone for HLR & NOD
      phone: veteran.mobilePhone,
    },
  };
  contactInfoValidation(errors, _fieldData, newFormData);
};
```

Reply:
agree that using `contactInfoValidation` makes it less readable... maybe instead you could add some smaller helper functions that validate the common fields we check in both of the validation functions? something like:

```
const validateBaseContactInfo = (phoneNumber, email) => {
  // if statements for mobile phone and email
}

const validateAddressInfo = (address) => {
  // if statements for address validations
}
```
then you could call each of them as needed in each of your validation functions like:
```
export const contactInfoValidation = (errors = {}, _fieldData, formData) => {
  validateBaseContactInfo(...)
  if (!homeless) {
    validateAddressInfo(...)
  }
}
```